ldblink.cpp

```diff
@@ -438,9 +425,16 @@ class DBLinkFactory : public TransformFunctionFactory
 
 		// Read Params:
 		ParamReader params = srvInterface.getParamReader();
-		if( params.containsParameter("cid") ) {
+		if( params.containsParameter("cid") ) {				// Start checking "cid" param
 			cid = params.getStringRef("cid").str() ;
-		} else if( params.containsParameter("connect") ) {
+		} else if( params.containsParameter("connect") ) {	// if "cid" is undef try with "connect"
+			connect = true ;
+			cid = params.getStringRef("connect").str() ;
+		} else if (srvInterface.getUDSessionParamReader("library").containsParameter("connect_secret")) {
+															// if both "cid" and "connect" are not defined try "connect_secret" session
+			connect = true ;
+			cid = srvInterface.getUDSessionParamReader("library").getStringRef("connect_secret").str() ;
```

Comment:
@mfelici so there are two params connect and connect_secret? I don't really get it, if I use the connect_secret param then it will not be displayed in the logs?

Reply:
@oualib Yes. There are two parameters but they belogs to different categories:

- CONNECT (like CID) is a **UDx parameter**. Meaning that you have to specify them in the UDx call. For example `SELECT DBLINK(USING PARAMETERS cid='blah'...)` or `SELECT DBLINK(USING PARAMETER connect='blah'...)`. **These parameter will be recorded in query_requests**;
- CONNECT_SECRET is a **Session parameter**! You set the session parameter before using DBLINK via SQL (`ALTER SESSION SET UDPARAMETER...`) and it will remain valid for all subsequent DBLINK calls **in the same session**. See the examples [here](https://github.com/mfelici/dblink#alternative-connection-definition-methods). **The value of this parameter won't be recorded in query_requests.**

Reply:
In light of some coming features in Vertica, I think we should change the way we do this.  I sent you a DM.

Reply:
- `cid` and `connect` as UDx parameter are not impacted by the upcoming (?) changes. 
- `connect_secret` as Session Parameter could eventually change its name to `connect` in case future Vertica releases won't hide anymore variables ending with _secret.